tests/e2e/per_site_exporter_test.go

```diff
@@ -0,0 +1,357 @@
+					return fmt.Errorf("expected 'OK', got '%s'", string(body))
+				}
+				return nil
+			}, timeout, interval).Should(Succeed())
+		})
+	})
+
+	Describe("Per-Site Metrics with Traffic", func() {
+		var (
+			testServer    *testhelpers.ProxyTestServer
+			client        *http.Client
+			metricsClient *http.Client
+		)
+
+		BeforeEach(func() {
+			podIP, err := getPodIP()
+			Expect(err).NotTo(HaveOccurred(), "Failed to get pod IP")
+
+			var err2 error
+			testServer, err2 = testhelpers.NewProxyTestServer("Per-site exporter test", podIP, 0)
+			Expect(err2).NotTo(HaveOccurred(), "Failed to create test server")
+
+			client, err2 = testhelpers.NewSquidProxyClient(serviceName, namespace)
+			Expect(err2).NotTo(HaveOccurred(), "Failed to create proxy client")
+
+			metricsClient = newHTTPSClient(10 * time.Second)
+		})
+
+		AfterEach(func() {
+			if testServer != nil {
+				testServer.Close()
+			}
+		})
+
+		getPerSiteMetricsValue := func(metricsContent, metricName, hostname string) (float64, error) {
+			lines := strings.Split(metricsContent, "\n")
+			pattern := fmt.Sprintf(`^%s\{.*hostname="%s".*\}\s+([0-9.]+)`, regexp.QuoteMeta(metricName), regexp.QuoteMeta(hostname))
+			re := regexp.MustCompile(pattern)
+			for _, line := range lines {
+				if matches := re.FindStringSubmatch(line); len(matches) >= 2 {
+					value, err := strconv.ParseFloat(matches[1], 64)
+					if err == nil {
+						return value, nil
+					}
+				}
+			}
+			return 0, fmt.Errorf("metric %s for hostname %s not found", metricName, hostname)
+		}
+
+		getPerSiteMetrics := func() (string, error) {
+			metricsURL := fmt.Sprintf("https://%s.%s.svc.cluster.local:9302/metrics", serviceName, namespace)
+			resp, err := metricsClient.Get(metricsURL)
+			if err != nil {
+				return "", err
+			}
+			defer resp.Body.Close()
+			if resp.StatusCode != http.StatusOK {
+				return "", fmt.Errorf("per-site metrics endpoint returned status %d", resp.StatusCode)
+			}
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				return "", err
+			}
+			return string(body), nil
+		}
+
+		It("should track per-site request metrics when traffic flows through proxy", func() {
+			testHostname := strings.Split(strings.TrimPrefix(testServer.URL, "http://"), ":")[0]
+			testURL := testServer.URL + "?" + generateCacheBuster("per-site-metrics-test")
+
+			By("Making HTTP requests through the proxy")
+			for i := 0; i < 3; i++ {
+				resp, _, err := testhelpers.MakeProxyRequest(client, testURL+fmt.Sprintf("&req=%d", i))
+				Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Request %d should succeed", i))
+				resp.Body.Close()
+			}
+
+			time.Sleep(5 * time.Second)
+
+			Eventually(func() bool {
+				metricsContent, err := getPerSiteMetrics()
+				if err != nil {
+					return false
+				}
+				requestMetric, err := getPerSiteMetricsValue(metricsContent, "squid_site_requests_total", testHostname)
+				if err != nil {
+					return false
+				}
+				return requestMetric >= 3
```

Comment:
I think it would have been more robust to capture the output for the same `getPerSiteMetricsValue` call before generating the traffic and then verifying that `requestMetric` minus the original is at least 3.